File: Threshold/threshold.cpp
#include  <opencv2/opencv.hpp>
#include <iostream>

using namespace std;
using namespace cv;

int main(int c,char **argv)
{
	Mat img = imread(argv[1],1);
	if(img.empty())
	{
		cout<<"Error,image can't be read"<<endl;
		return 0;
	}
	Mat gray,thresh_binary,thresh_inv,thresh_trunc,thresh_tozero;
	cvtColor(img,gray,CV_BGR2GRAY);
	threshold(gray,thresh_binary,127,255,CV_THRESH_BINARY);
	threshold(gray,thresh_inv,127,255,CV_THRESH_BINARY_INV);
	threshold(gray,thresh_trunc,200,255,CV_THRESH_TRUNC);
	threshold(gray,thresh_tozero,200,255,CV_THRESH_TOZERO);

	namedWindow("thresh_binary",WINDOW_NORMAL);
	imshow("thresh_binary",thresh_binary);
	namedWindow("thresh_inv",WINDOW_NORMAL);
	imshow("thresh_inv",thresh_inv);
	namedWindow("thresh_trunc",WINDOW_NORMAL);
	imshow("thresh_trunc",thresh_trunc);
	namedWindow("thresh_tozero",WINDOW_NORMAL);
	imshow("thresh_tozero",thresh_tozero);
	waitKey(0);
}

File: Threshold/threshold.py
import cv2 as cv
import numpy as np

image = cv.imread('gradient.png',0)
ret,thresh1 = cv.threshold(image,127,255,cv.THRESH_BINARY)
ret,thresh2 = cv.threshold(image,127,255,cv.THRESH_BINARY_INV)
ret,thresh3 = cv.threshold(image,127,255,cv.THRESH_TRUNC)
ret,thresh4 = cv.threshold(image,127,255,cv.THRESH_TOZERO)
ret,thresh5 = cv.threshold(image,127,255,cv.THRESH_TOZERO_INV)


cv2.namedWindow("thresh1",cv2.WINDOW_NORMAL)
cv2.imshow("thresh1",thresh1)
cv2.namedWindow("thresh2",cv2.WINDOW_NORMAL)
cv2.imshow("thresh2",thresh2)
cv2.namedWindow("thresh3",cv2.WINDOW_NORMAL)
cv2.imshow("thresh3",thresh3)
cv2.namedWindow("thresh4",cv2.WINDOW_NORMAL)
cv2.imshow("thresh4",thresh4)
cv2.waitKey(0)